gitpod-io/gitpod
# [ws-daemon] Introduce IO limiting

## Description
This PR
- is joint work together with @Furisto,
- restructures the cgroup handling within ws-daemon,
- introduces IO bandwidth limiting for workspaces

## How to test
- Enable the IO limiting in the experimental section of the installer
- Deploy a cluster and start a workspace
- Check the `io.max` file of the workspace's container. It should contain entries for all devices.

## Release Notes

```release-note
Add IO bandwidth limit support
```
